ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -364,15 +354,16 @@ public StreamPullQueryMetadata createStreamPullQuery(
             .putAll(statementOrig.getSessionConfig().getOverrides())
             .put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
             .put(StreamsConfig.NUM_STREAM_THREADS_CONFIG, 1)
```

Comment:
Is this important for threadsafety in your processor?  Might make sense to comment on that here if so.

Reply:
No, I just think it'd be wasteful to spend more than one thread on this operation.

Reply:
Turns out, I did have a comment explaining the full set of configs a few lines above, but I'd already forgotten to update it when I removed the commit interval, which isn't a good sign.

I moved the comments down to explain each config option independently, so this is hopefully clearer now.